pipeline/console/environments.py

```diff
@@ -84,87 +94,75 @@ def _create_environment(name: str, packages: List[str]) -> EnvironmentGet:
     return EnvironmentGet.parse_obj(environment_dict)
 
 
-def _delete_environment(name_or_id: str, by_name=False) -> None:
+def _delete_environment(id: str) -> None:
+    """Delete an environment by id. For deletion by name, resolve the environment ID
+    first."""
     remote_service = PipelineCloud(verbose=False)
     remote_service.authenticate()
 
-    environment_information = _get_environment(name_or_id, by_name=by_name)
-    remote_service._delete(f"/v2/environments/{environment_information.id}")
+    remote_service._delete(f"/v2/environments/{id}")
 
 
 def _update_environment(
-    name_or_id: str,
+    id: str,
     *,
     locked: bool = None,
     python_requirements: List[str] = None,
-    by_name=False,
 ) -> EnvironmentGet:
+    """Update an environment by id. For update by name, resolve the environment ID
+    first."""
     if locked is None and python_requirements is None:
         raise Exception(
             "Must un/lock or define python_requirements when updating an Environment"
         )
     remote_service = PipelineCloud(verbose=False)
     remote_service.authenticate()
 
-    environment_information = _get_environment(name_or_id, by_name=by_name)
-
     update_schema = EnvironmentPatch(
         python_requirements=python_requirements, locked=locked
     )
     response = remote_service._patch(
-        f"/v2/environments/{environment_information.id}",
+        f"/v2/environments/{id}",
         update_schema.dict(),
     )
     return EnvironmentGet.parse_obj(response)
 
 
 def _add_packages_to_environment(
-    name_or_id: str,
+    environment: EnvironmentGet,
```

Comment:
I like the way you pass in this environment object!

Reply:
Yeah, made sense since we're fetching it off the bat and needed other non-id properties in the method.